lib/Models/ArcGisMapServerCatalogItem.js

```diff
@@ -178,28 +180,43 @@ defineProperties(ArcGisMapServerCatalogItem.prototype, {
 var partsRegex = new RegExp('^(.*\/MapServer\/)([0-9]+)', 'i');
 
 ArcGisMapServerCatalogItem.prototype._load = function() {
+    var that = this;
+    function getJson(uri) {
+        return loadJson(proxyCatalogItemUrl(that, uri.addQuery('f', 'json').toString()));
+    }
     if (!defined(this._mapServerData) || !defined(this._layersData)) {
-        var reparts = partsRegex.exec(this.url);
-        var serviceUrl, layersUrl;
-        if (reparts) {
-            // URL is a single REST layer, like .../arcgis/rest/services/Society/Society_SCRC/MapServer/16
-            serviceUrl = cleanAndProxyUrl(this, reparts[1]) + '?f=json';
-            layersUrl = cleanAndProxyUrl(this, this.url) + '?f=json';
-        } else {
-            serviceUrl = cleanAndProxyUrl(this, this.url) + '?f=json';
-            layersUrl = cleanAndProxyUrl(this, this.url) + '/layers?f=json';
+        var uri = new URI(this.url);
+        var baseUri = uri;
+        var layers = 'layers';
+        if (uri.segment(-1).match(/\d+/)) {
+            // URL is a single REST layer, like .../arcgis/rest/services/Society/Society_SCRC/MapServer/16            
+            baseUri = uri.clone().segment(-1, '');
+            layers = uri.segment(-1);
+
+            this.layers = layers; // ## is this ok to do?
         }
-
-        var serviceMetadata = this.mapServerData || loadJson(serviceUrl);
-        var layersMetadata = this.mapServerLayersData || loadJson(layersUrl);
-
-        var that = this;
-        return when.all([serviceMetadata, layersMetadata]).then(function(results) {
+        var serviceUri = baseUri.clone();
+        var layersUri = baseUri.clone().segment(layers); // either 'layers' or a number
+        var legendUri = baseUri.clone().segment('legend');
+
+        var serviceMetadata = this.mapServerData || getJson(serviceUri);
+        var layersMetadata = this.mapServerLayersData || getJson(layersUri);
+        var legendData = getJson(legendUri).
+            then(this.loadLegendFromJson.bind(this)).
```

Comment:
You should only do this if the catalog item doesn't already have a `legendUrl`.  If one is supplied in the init file, it should take precedence over an automatically generated one.